python/cudf/cudf/_lib/io/datasource.pyx

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from cudf._lib.cpp.io.types cimport datasource
+
+cdef class Datasource:
+    cdef datasource* get_datasource(self) except *:
```

Comment:
```suggestion
    cdef datasource* get_datasource(self) nogil except *:
```

Reply:
according to Cython `raise` is valid to use in nogil, but not sure if constructing a `NotImplementedError` is valid or not. @jdye64 if Cython errors on this just need to add a `with gil:` context before the `raise` call.

https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#releasing-the-gil